test/test_binance_market.py

```diff
@@ -323,6 +323,29 @@ def test_cancel_all(self):
         for cr in cancellation_results:
             self.assertEqual(cr.success, True)
 
+    def test_order_price_precision(self):
+        symbol = "IOSTETH"
+        bid_price: float = self.market.get_price(symbol, True)
+        ask_price: float = self.market.get_price(symbol, False)
+        amount: float = 0.02 / bid_price
+        quantized_amount: Decimal = self.market.quantize_order_amount(symbol, amount)
+        self.market.buy("IOSTETH", amount)
+
+        # Intentionally setting invalid price to prevent getting filled
+        quantize_bid_price: Decimal = self.market.quantize_order_price(symbol, bid_price * 0.7)
+        quantize_bid_price -= Decimal('0.00000005')
+        quantize_ask_price: Decimal = self.market.quantize_order_price(symbol, ask_price * 1.5)
+        quantize_ask_price += Decimal('0.00000005')
+
+        self.market.buy(symbol, quantized_amount, OrderType.LIMIT, quantize_bid_price)
```

Comment:
This is a bad test case. The buy and sell functions are supposed to have floating point inputs because the strategies shouldn't care about these market details. What you should do is to give it a floating point number that goes beyond the required precision (i.e. an invalid order) and test that the market connector would make it a valid order again.

Reply:
e.g. Binance API specifies filter rules about what the minimum price and order size increments are for every trading pair. So for IOSTETH the minimum price tick is 1e-8 ETH and the minimum order size quantum is 1 IOST.

So the expectation here is, you should be able to give the Binance market connector an order to buy 5.012 IOST at 1.2321e-7 ETH and it'll quantize the size to 5 IOST and the price to 0.00000012. And you should check that the order went through.